db.go

```diff
@@ -771,6 +771,13 @@ func (db *DB) EnableCompactions() {
 	level.Info(db.logger).Log("msg", "compactions enabled")
 }
 
+// Trigger compactions
+func (db *DB) Compact() error {
+	db.autoCompactMtx.Lock()
```

Comment:
I can't be sure to make the `db.compact()` public is correct, because if I do, I enable someone to use `db.Compact()` without disabling auto compact. Is it more safe to add additional `db.Compact()` and call the `db.DisableCompactions()` and `defer db.EnableCompactions()` inside it?

Reply:
No 2 compactions can run in parallel. So even if you run `db.compact()` manually and from auto compaction, they would still be synchronised by `db.cmtx`.

Cases:
* Holding `autoCompactMtx` is as good as directly calling `db.compact()` directly, because of `cmtx`.
* Calling `db.EnableCompactions()` is similar to above, but will only delay the next auto compaction in some cases (this is not a problem, as the manual call takes the place of auto compaction).

PS: I think exposing `db.Compact()` is not a good idea in general. Would manually using a compactor (`LeveledCompactor`) in CLI and doing the compaction make more sense? (would mean some code duplication)

/cc @krasi-georgiev

Reply:
I couldn't understand the reason behind not using `db.Compact()` directly because I thought as it will be used for offline scenario. Is the concern not enabling anyone to trigger compaction other than using CLI?

Reply:
I think exposing `Compact()` should be fine as tsdb is internal to prometheus. My previous thought was that exposing means it can be misused. But as its internal, should be fine.

Any thoughts on this @krasi-georgiev (and about disabling compactions inside `db.Compact()`)?

Reply:
Exposing it in a user-facing tool is not internal. In fact the command as proposed here is relatively dangerous, as it won't have the appropriate block sizes.